src/ControlSystem/ControlErrors/SizeControlStateAhSpeed.cpp

```diff
@@ -0,0 +1,112 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/ControlErrors/SizeControlStateAhSpeed.hpp"
+
+#include <cmath>
+
+namespace SizeControlStates {
+void AhSpeed::update(const gsl::not_null<SizeControlInfo*> info,
+                     const SizeControlStateUpdateArgs& update_args,
+                     const CrossingTimeInfo& crossing_time_info) const {
+  const double min_char_speed = update_args.min_char_speed;
+  const double min_comoving_char_speed = update_args.min_comoving_char_speed;
+
+  // Note that delta_radius_is_in_danger and char_speed_is_in_danger
+  // can be different for different SizeControlStates.
+  //
+  // The value of 20 causes the control system to panic easily if
+  // delta_radius is decreasing quickly.
```

Comment:
Could you give a little more info on this number? Maybe how you came to choose it (even if it was just experimentation in SpEC)? That way in a year when I'm looking back at this I don't just see a magic `20.0` that says "this is the right value".